src/main/java/gregtech/api/util/InventoryUtils.java

```diff
@@ -115,4 +128,33 @@ private static void mergeItemStacks(Collection<ItemStack> source, Collection<Ite
                 }
         }
     }
+
+    /**
+     * Computes a resultant number of ItemStacks from an ItemStack exceeding its
+     * {@link net.minecraft.item.ItemStack#getMaxStackSize()}.<br />
+     * Resultant ItemStacks will be added to {@code collapsedStack} as full stacks followed
+     * by a partial stack if applicable.<br />
+     *
+     * @param stack      the ItemStack exceeding its maximum stack size.
+     * @param collapsedStack A Collection of ItemStacks for the resultant ItemStacks to be merged into.
+     */
+    public static void computeItemStacks(Collection<ItemStack> collapsedStack, ItemStack stack) {
+        int overCount  = stack.getCount();
+        int maxCount = stack.getMaxStackSize();
+
+        int numStacks = overCount / maxCount;
+        int remainder = overCount % maxCount;
+
+        for(int fullStackCount = numStacks; fullStackCount > 0; fullStackCount--) {
+            ItemStack fullStack = stack.copy();
+            fullStack.setCount(maxCount);
+            collapsedStack.add(fullStack);
+        }
+
+        if(remainder > 0) {
+            ItemStack partialStack = stack.copy();
+            partialStack.setCount(remainder);
+            collapsedStack.add(partialStack);
+        }
+    }
```

Comment:
I don't like implementation of this. Because you need to pass in collection which gets modified as side effect.
I would rather see this as function taking in "big size" ItemStack and returning ArrayList of "normal size" ItemStacks. Needed to say this should be also handle situation if "normal size" ItemStack or empty one is passed in.

This way it will be more readable (or at least I think so). And you could also use it in case like https://github.com/GregTechCE/GregTech/blob/master/src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityMultiFurnace.java#L217